src/Forms/Shared/Samples/Symbology/CustomDictionaryStyle/CustomDictionaryStyle.xaml.cs

```diff
@@ -30,45 +32,83 @@ public partial class CustomDictionaryStyle : ContentPage
         // Uri for the restaurants feature service.
         private readonly Uri _restaurantUri = new Uri("https://services2.arcgis.com/ZQgQTuoyBrtmoGdP/arcgis/rest/services/Redlands_Restaurants/FeatureServer/0");
 
+        // Hold a reference to the custom dictionary styles for use in the event handlers. 
+        private DictionarySymbolStyle _localRestaurantStyle;
+        private DictionarySymbolStyle _restaurantStyleFromPortal;
+
+        // Hold a reference to the feature layer for use in the event handlers.
+        private FeatureLayer _restaurantLayer;
+
         public CustomDictionaryStyle()
         {
             InitializeComponent();
-            Initialize();
+            _ = Initialize();
         }
 
-        private async void Initialize()
+        private async Task Initialize()
         {
             try
             {
-                // Create a new map with a streets basemap.
-                Map map = new Map(BasemapStyle.ArcGISStreets);
+                // Create a new map with a topographic basemap.
+                Map map = new Map(BasemapStyle.ArcGISTopographic);
 
                 // Create the restaurants layer and add it to the map.
-                FeatureLayer restaurantLayer = new FeatureLayer(_restaurantUri);
-                map.OperationalLayers.Add(restaurantLayer);
+                _restaurantLayer = new FeatureLayer(_restaurantUri);
+                map.OperationalLayers.Add(_restaurantLayer);
 
                 // Load the feature table for the restaurants layer.
-                await restaurantLayer.FeatureTable.LoadAsync();
+                await _restaurantLayer.FeatureTable.LoadAsync();
 
-                // Open the custom style file.
-                DictionarySymbolStyle restaurantStyle = await DictionarySymbolStyle.CreateFromFileAsync(_stylxPath);
+                // Open the local custom style file.
+                _localRestaurantStyle = await DictionarySymbolStyle.CreateFromFileAsync(_stylxPath);
 
-                // Create the dictionary renderer with the style file and the field overrides.
-                DictionaryRenderer dictRenderer = new DictionaryRenderer(restaurantStyle);
+                // Open a portal item containing a dictionary symbol style.
+                ArcGISPortal portal = await ArcGISPortal.CreateAsync();
+                PortalItem portalItem = await PortalItem.CreateAsync(portal, "adee951477014ec68d7cf0ea0579c800");
 
-                // Set the restaurant layer renderer to the dictionary renderer.
-                restaurantLayer.Renderer = dictRenderer;
+                // Open the portal custom style file.
+                _restaurantStyleFromPortal = await DictionarySymbolStyle.OpenAsync(portalItem);
 
                 // Set the map's initial extent to that of the restaurants.
-                map.InitialViewpoint = new Viewpoint(restaurantLayer.FullExtent);
+                map.InitialViewpoint = new Viewpoint(_restaurantLayer.FullExtent);
```

Comment:
I wonder if it would be better to re-order this code to group the map manipulation at the start, then set up the styles and renderers after.